lib/server.go

```diff
@@ -2939,19 +2981,13 @@ func (srv *Server) Start() {
 		go srv.miner.Start()
 	}
 
-	srv.networkManager.Start()
-
-	// On testnet, if the node is configured to be a PoW block producer, and it is configured
-	// to be also a PoS validator, then we attach block mined listeners to the miner to kick
-	// off the PoS consensus once the miner is done.
-	if srv.params.NetworkType == NetworkType_TESTNET && srv.fastHotStuffConsensus != nil {
-		tipHeight := uint64(srv.blockchain.blockTip().Height)
-		if srv.params.IsFinalPoWBlockHeight(tipHeight) || srv.params.IsPoSBlockHeight(tipHeight) {
-			if err := srv.fastHotStuffConsensus.Start(); err != nil {
-				glog.Errorf("NewServer: Error starting fast hotstuff consensus %v", err)
-			}
-		}
+	// Initialize state syncer mempool job, if needed.
+	if srv.stateChangeSyncer != nil {
```

Comment:
@superzordon what's the best place to trigger the state syncer mempool sync routine? With PoS, it's possible for a node to start up, no sync blocks from a peer, and instead join the validator set directly.

It seems cleanest to decouple this mempool sync routine from the _startSync operation where it originally was. I have no strong opinion on this, so LMK if this is an approprate place to move this to

Reply:
It belongs in server.Start - putting it in sync is incorrect and should already be addressed in main (or Z has a PR for it).